### PR TITLE
Used mock.Mock() in SMTPBackendTests.

### PR DESCRIPTION
**Pull request by kalekseev**
We are typechecking mail tests now https://github.com/typeddjango/django-stubs/pull/208 and this patch would help us to pass tests without ignoring errors. Not sure would you accept such patches to test modules, let me know if I have to create ticket for it.

### CODE REVIEW

File: tests/mail/tests.py [hunk 1]
```diff
@@ -46,6 +46,16 @@ def assertMessageHasHeaders(self, message, headers):
                         'the following headers: %s' % (headers - msg_headers),)
 
 
+class FakeSMTPConnection(SMTP):
+    """
+    A SMTP connection that doesn't call any network services on initialization.
+    Helps to simulate an already open connection.
+    """
+
+    def __init__(self, host='', port=0, local_hostname=None, **kwargs):
```
Comment (codingjoe) [inline tests/mail/tests.py]: Why do you accept argument here, if you don't pass them to super.
Maybe you can just the the correct default for `local_hostname` here and pass the variable.

```suggestion
    def __init__(self, host='', port=0, local_hostname='[127.0.0.1], **kwargs):
        super()__init__(host=host, port=port, local_hostname=local_hostname, **kwargs)
```

Comment (kalekseev) [reply tests/mail/tests.py]: I don't think that fakeness of the object should depend on args provided to it. If we accept your suggestion then someone can call that fake class with random host arg and think that it still fake while it not anymore.

Maybe poping from kwargs will look less confusing?

```
def __init__(self, **kwargs):
    kwargs.pop('host')
    kwargs.pop('localhost_name')
    super().__init__(host='', local_hostname='[127.0.0.1], **kwargs)
```

Comment (codingjoe) [reply tests/mail/tests.py]: It looks strange and sounds strange, but I see your point. Yes, it's fake and should be fake even with real input. Maybe we can actually leave it as is, but with a comment?

**Comment by felixxm** [comment]
@kalekseev Thanks :+1: Do we need `FakeSMTPConnection`? I would use `backend.connection = mock.Mock()` in these tests.

**Comment by kalekseev** [comment]
> Do we need `FakeSMTPConnection`? I would use `backend.connection = mock.Mock()` in these tests.

@felixxm it mostly developer's choice, I prefer fakes because they give me a real interface while mock happily accepts any calls to it, more details: [link1](https://nedbatchelder.com/blog/201206/tldw_stop_mocking_start_testing.html), [link2](https://pythonspeed.com/articles/verified-fakes/). In this case tests doesn't call connection object, otherwise `connection = True` would fail, so mock is good enough. Another reason was the effort to make tests pass mypy typecheck, boolean instead of SMTPConnection was clear type violation but mock should work. 

So I'm okay with mocks too, let me know if I should update pr with mock version.

**Comment by felixxm** [comment]
> In this case tests doesn't call connection object, otherwise connection = True would fail, so mock is good enough.

Exactly, we don't really use `connection` in these tests, I think `object()` for a fake connection would be even better than `Mock()`. If it's fine for `mypy` typecheck, I would change `FakeSMTPConnection()` to `object()`.

**Comment by kalekseev** [comment]
>  If it's fine for mypy typecheck, I would change FakeSMTPConnection() to object()

Bare object won't pass mypy typecheck, I updated pr with `mock.Mock(spec=object())`

**Comment by felixxm** [comment]
@kalekseev Thanks :+1:

**Pull request merged**